solution/index.js

```diff
@@ -0,0 +1,870 @@
+// this will display the context menu.
+function contextClickHandler(event) {
+  const prevMenu = document.querySelector('.context-menu')
+  if (prevMenu) prevMenu.remove()
+
+  const target = event.target.closest('LI')
+
+  if (!target) return
+
+  event.preventDefault() // to prevent the browser default context menu.
+
+  const menuOptionDelete = createMenuOption('delete task', removeTask)
+  const contextMenu = createElement('div', [menuOptionDelete], ['context-menu'])
+
+  function removeTask() {
+    const taskId = getLiTaskId(target)
+    deleteTask(taskId)
+  }
+
+  // this function is used to create menu options. feel free to add one of your own!
+  function createMenuOption(text, handler) {
+    const menuOption = createElement('div', [`${text}`], [`${text.replace(/[' ']/g, '-')}-option`])
+    menuOption.addEventListener('click', handler)
+    return menuOption
+  }
+
+  contextMenu.style.position = 'absolute'
+  contextMenu.style.zIndex = 1000
+  document.body.append(contextMenu)
+
+  contextMenu.style.left = event.pageX + 'px'
+  contextMenu.style.top = event.pageY + 'px'
+
+  document.addEventListener('click', exitContextMenuHandler)
+
+  function exitContextMenuHandler(event) {
+    contextMenu.remove()
+    document.removeEventListener('onclick', exitContextMenuHandler)
+  }
+}
+
+// this is a event delegation handler for left click events on buttons
+function clickEventHandler(event) {
+  const target = event.target.closest('BUTTON')
+
+  if (!target) return
+
+  // Each html button element has a dataset role.
+  // this way we determine the action to execute.
+  const targetRole = target.dataset.role
+
+  if (targetRole === 'adding-task') {
+    const inputValue = target.parentElement.querySelector('input').value
+    if (inputValue === '') {
+      alert('Cant add an empty task!')
+    } else {
+      const relevantListId = getAncestorSectionListId(target)
+      addTask(relevantListId, inputValue)
+    }
+  } else if (targetRole === 'adding-list') {
+    const inputValue = document.getElementById('add-new-list').value
+    if (inputValue === '') {
+      alert('Cant add an empty task!')
+    } else {
+      addNewList(inputValue)
+    }
+  } else if (targetRole === 'saving-board') {
+    putTasksToApi().catch((error) => alert(error))
+    startLoadAnimation()
+    setTimeout(() => {
+      renderBoard()
+    }, 100)
+  } else if (targetRole === 'loading-board') {
+
+
+    startLoadAnimation()
+
+    getTasksFromApi()
+      .then((tasks) => {
+        updateLocalStorageTasksInNativeFormat(tasks)
+        renderBoard()
+      })
+      .catch((error) => alert(error))
+  } else if (targetRole === 'delete-list') {
+    let relevantListId = getAncestorSectionListId(target)
+    deleteList(relevantListId)
+  }
+}
+
+let mouseInTask = null // this is for monitoring if the mouse is pointing at the task
+
+function altKeyDownEventHandler(event) {
+  const keyPressed = event.key
+
+  if (mouseInTask && keyPressed === 'Alt') {
```

Comment:
There's a better way to see if ALT was pressed, that doesn't require more event handlers